graphql_compiler/schema/__init__.py

```diff
@@ -340,36 +340,47 @@ def _serialize_date(value: Any) -> str:
 
 def _parse_date_value(value: Any) -> date:
     """Deserialize a Date object from its proper ISO-8601 representation."""
-    return arrow.get(value, "YYYY-MM-DD").date()
+    if type(value) == date:
+        # We prefer exact type equality instead of isinstance() because datetime objects
+        # are subclasses of date but are not interchangeable for dates for our purposes.
+        return value
+    elif isinstance(value, str):
+        return arrow.get(value, "YYYY-MM-DD").date()
```

Comment:
Any need to assert that value follows ISO-8601 prior to parsing / truncating?

```In [7]: arrow.get('2010-01-01 2020-01-01 boom boom', 'YYYY-MM-DD')                   
Out[7]: <Arrow [2010-01-01T00:00:00+00:00]>
```

Reply:
This is a good point. Maybe I should just use the Python builtins for this case, since there are many fewer subtleties in the date case than in the datetime case.

Unfortunately, `arrow` in general is really permissive with its inputs, and I don't immediately know of a good alternative for ISO 8601 datetime parsing. If you have one you like, I'm happy to eliminate `arrow` as a dependency altogether and switch to it entirely.

Reply:
Maybe this one?
https://pypi.org/project/ciso8601/

Also @branen what do you think? Is it worth ripping out `arrow` entirely as a dependency and switching to `ciso8601` instead?

Reply:
@obi1kenobi, I'd like to give this some more thought before I express an opinion about `arrow` vs `ciso8601` vs anything else, but I do want to note this, which is disappointing, given its purported support for ISO-8601:

```python
>>> arrow.get('2020-W02')
<Arrow [2020-01-01T00:00:00+00:00]>
```

(It should be 2020-01-08.)

Reply:
I read through `ciso8601` code and poked at it a little, and it looks great.  I definitely think we should use it in place of `arrow`.

That said, since we only support a subset ISO-8601, it'd be nice to phrase the error messages we use in a way that reflect this.  `ciso8601` implements something pretty close to RFC 3339, so I'll suggest some deltas that reference it.